File: engine/src/flutter/impeller/renderer/command.cc
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "impeller/renderer/command.h"

#include "impeller/base/validation.h"
#include "impeller/core/formats.h"

namespace impeller {

bool Command::BindVertices(const VertexBuffer& buffer) {
  if (buffer.index_type == IndexType::kUnknown) {
    VALIDATION_LOG << "Cannot bind vertex buffer with an unknown index type.";
    return false;
  }

  vertex_buffers = {buffer.vertex_buffer};
  vertex_buffer_count = 1u;
  element_count = buffer.vertex_count;
  index_buffer = buffer.index_buffer;
  index_type = buffer.index_type;
  return true;
}

}  // namespace impeller

File: engine/src/flutter/tools/engine_tool/lib/src/commands/command.dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:args/args.dart';
import 'package:args/command_runner.dart';

import '../environment.dart';

/// The base class that all commands and subcommands should inherit from.
abstract base class CommandBase extends Command<int> {
  /// Constructs the base command.
  CommandBase({
    required this.environment,
    this.help = false,
    int? usageLineLength,
  }) : argParser = ArgParser(usageLineLength: usageLineLength);

  /// The host system environment.
  final Environment environment;

  /// Whether the Command is being constructed only to print the usage/help
  /// message.
  final bool help;

  @override
  final ArgParser argParser;

  @override
  void printUsage() {
    environment.logger.status(usage);
  }
}
